Sources/protoc-gen-grpc-swift/Generator-Client.swift

```diff
@@ -226,6 +289,73 @@ fileprivate extension Generator {
   }
 }
 
+extension Generator {
+  fileprivate var methodArguments: [String] {
+    switch self.streamType {
+    case .unary:
+      return [
+        "_ request: \(self.methodInputName)",
+        "callOptions: CallOptions"
+      ]
+    case .serverStreaming:
+      return [
+        "_ request: \(self.methodInputName)",
+        "callOptions: CallOptions",
+        "handler: @escaping (\(methodOutputName)) -> Void"
+      ]
+
+    case .clientStreaming:
+      return ["callOptions: CallOptions"]
+
+    case .bidirectionalStreaming:
+      return [
+        "callOptions: CallOptions",
+        "handler: @escaping (\(methodOutputName)) -> Void"
+      ]
+    }
+  }
+
+
+  fileprivate var methodArgumentsWithoutCallOptions: [String] {
```

Comment:
To avoid the risk of getting things wrong here, should this be defined programmatically in terms of `methodArguments` by delegating to `methodArguments` and stripping the one that starts with `callOptions:`?

Reply:
Yeah, good call.